src/components/accession2/QuantityModal.tsx

```diff
@@ -0,0 +1,150 @@
+import { Accession2, updateAccession2 } from 'src/api/accessions2/accession';
+import useForm from 'src/utils/useForm';
+import { ServerOrganization } from 'src/types/Organization';
+import { Unit, WEIGHT_UNITS_V2 } from '../seeds/nursery/NewTest';
+import useSnackbar from 'src/utils/useSnackbar';
+import CalculatorModal from './CalculatorModal';
+
+export interface QuantityDialogProps {
+  open: boolean;
+  setOpen: () => void;
+  accession: Accession2;
+  onClose: () => void;
+  reload: () => void;
+  organization: ServerOrganization;
+}
+
+export default function QuantityDialog(props: QuantityDialogProps): JSX.Element {
+  const { onClose, open, accession, reload, organization, setOpen } = props;
+
+  const [record, setRecord, onChange] = useForm(accession);
+  const [isCalculatorOpened, setIsCalculatorOpened] = useState(false);
+  const theme = useTheme();
+  const snackbar = useSnackbar();
+
+  const saveQuantity = async () => {
+    const response = await updateAccession2(record);
+    if (response.requestSucceeded) {
+      reload();
+      onCloseHandler();
+    } else {
+      snackbar.toastError();
+      setRecord(accession);
+      onCloseHandler();
+    }
+  };
+
+  const onChangeRemainingQuantity = (id: string, value: number) => {
+    if (record) {
+      if (id === 'seedsQuantity') {
+        setRecord({
+          ...record,
+          remainingQuantity: {
+            quantity: value,
+            units: 'Seeds',
+          },
+        });
+      } else {
+        setRecord({
+          ...record,
+          remainingQuantity: {
+            quantity: value,
+            units: record.remainingQuantity?.units === 'Seeds' ? 'Grams' : record.remainingQuantity?.units || 'Grams',
+          },
+        });
+      }
+    }
+  };
+
+  const onChangeUnit = (newValue: Unit) => {
+    if (record) {
+      setRecord({
+        ...record,
+        remainingQuantity: {
+          quantity: record.remainingQuantity?.quantity || 0,
+          units: newValue.value,
+        },
+      });
+    }
+  };
+
+  const onCloseHandler = () => {
+    setRecord(accession);
+    onClose();
+  };
+
+  const openCalculator = () => {
+    setIsCalculatorOpened(true);
+    onClose();
+  };
+
+  return (
+    <>
+      <CalculatorModal
+        open={isCalculatorOpened}
+        onClose={() => setIsCalculatorOpened(false)}
+        record={record}
+        setRecord={setRecord}
+        onChange={onChange}
+        organization={organization}
+        reload={reload}
+        openPreviousModal={setOpen}
+      />
+      <DialogBox
+        onClose={onCloseHandler}
+        open={open}
+        title={strings.QUANTITY}
+        size='medium'
+        middleButtons={[
+          <Button label={strings.CANCEL} type='passive' onClick={onCloseHandler} priority='secondary' key='button-1' />,
+          <Button onClick={saveQuantity} label={strings.SAVE} key='button-2' />,
+        ]}
+      >
+        <Grid item xs={12} textAlign='left'>
```

Comment:
looks like this for me 
<img width="281" alt="Terraware App 2022-09-12 13-14-59" src="https://user-images.githubusercontent.com/1865174/189750388-05a08dbf-8f7c-4845-9d92-0d3d60f43139.png">

Reply:
Oh, my bad. I didn't push last changes :(